code/orphaned_procs/AStar.dm

```diff
@@ -123,23 +122,25 @@ Actual Adjacent procs :
 				continue
 
 			var/newg = cur.g + call(cur.source,dist)(T)
-			if(!T.PNode) //is not already in open list, so add it
-				open.Insert(new /PathNode(T,cur,newg,call(T,dist)(end),cur.nt+1))
+
+			var/PathNode/P = pnodelist[T]
```

Comment:
@optimumtact there is no bad index.
T is a turf, so it's a key, not an index.
and list[key_thats_not_in_there] is just null